crates/iceberg/src/writer/file_writer/track_writer.rs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::{
+    pin::Pin,
+    sync::{atomic::AtomicI64, Arc},
+};
+
+use tokio::io::AsyncWrite;
+
+use crate::io::FileWrite;
+
+/// `TrackWriter` is used to track the written size.
+pub(crate) struct TrackWriter {
+    inner: Box<dyn FileWrite>,
+    written_size: Arc<AtomicI64>,
```

Comment:
Writer only has `&mut` API so we can simplely use an `i64` here instead.

Reply:
The track writer will be pass to parquet writer and then we can't get the info on it. Thit written_size is used to share the value so that we can read it outside, like:
```
let written_size = ..;
let track_writer = TrackWriter(written_size.clone());
let parquet_writer = ParquetWriter(track_writer);

// We can use written_size to read the written size.
```